### PR TITLE
Update ResourceMgmtActionProvider.java

### LINKED ISSUE #2538: Refresh option not available for some resources that are not closed projects
- [X] I verified I can reproduce this issue against [latest Integration Build of Eclipse SDK](https://download.eclipse.org/eclipse/downloads/)

Issue was introduced in Eclipse IDE for Java Developers, Version: 2024-03 (4.31.0), Build id: 20240307-1437 ( https://www.eclipse.org/downloads/download.php?file=/technology/epp/downloads/release/2024-03/R/eclipse-java-2024-03-R-win32-x86_64.zip )

This situation is ***NOT*** seen in the previous quarterly release (Version: 2023-12 (4.30.0), Build id: 20231201-2043), and I don't see anything mentioned regarding this functionality @ https://eclipse.dev/eclipse/news/4.31/ .

The code change that caused this was https://github.com/eclipse-platform/eclipse.platform.ui/pull/1438. I submitted an initial fix for this @ https://github.com/eclipse-platform/eclipse.platform.ui/pull/2003 , however that fix did not cover all cases. 

I'm using Eclipse on Windows. 

## Steps to reproduce

A plugin that I work on has a view that extends org.eclipse.ui.navigator.CommonNavigator . In previous versions this provided 'Refresh (F5)' in the contextual menu for any resource. However now the menu option only exists for projects, but not other resources. 

Fairly easy to trigger / reproduce. Plug-in Spy tells me the following for the menu entry I'm talking about here:

oldEclipseOScript

## Community

- [X] I understand reporting an issue to this OSS project does not mandate anyone to fix it. Other contributors may consider the issue, or not, at their own convenience. The most efficient way to get it fixed is that I fix it myself and contribute it back as a good quality patch to the project.

### PR DESCRIPTION
**Pull request by DaveCarpeneto**
Fixes https://github.com/eclipse-platform/eclipse.platform.ui/issues/2538

### CODE REVIEW

File: bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java [hunk 1]
```diff
@@ -119,11 +119,11 @@ public void fillContextMenu(IMenuManager menu) {
...
 									// have builder
-
-		Iterator<IProject> projects = selectionToProjects(selection).iterator();
-
-		while (projects.hasNext() && (!hasOpenProjects || !hasClosedProjects || hasBuilder || isProjectSelection)) {
-			IProject project = projects.next();
+		List<IProject> projects = selectionToProjects(selection);
+		Iterator<IProject> projectIter = projects.iterator();
+		
+		while (projectIter.hasNext() && (!hasOpenProjects || !hasClosedProjects || hasBuilder || isProjectSelection)) {
```
Comment (jukzi) [inline bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java:125]: please use for-each loop instead

File: bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java [hunk 2]
```diff
@@ -148,7 +148,7 @@ public void fillContextMenu(IMenuManager menu) {
 		// Add the 'refresh' item if any selection is either (a) an open project, or (b)
                	// a non-project selection (so the 'refresh' item is not shown if all selections
                	// are closed projects)
-               	if (hasOpenProjects || !isProjectSelection) {
+		if (hasOpenProjects || projects.size() < selection.size()) {
```
Comment (jukzi) [inline bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java:151]: please add explanation why `projects.size() < selection.size()` is used here. That is not self-explaining

**Review by jukzi — CHANGES_REQUESTED** [review]
The commit message does not follow our best practice

File: bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java (hunk 2 above)
Comment (jukzi) [reply bundles/org.eclipse.ui.navigator.resources/src/org/eclipse/ui/internal/navigator/resources/actions/ResourceMgmtActionProvider.java:151]: maybe introduce a variable like selectionContainsNonProject

**Comment by merks** [comment]
> The commit message does not follow our best practice

You mean it should summarize what is being changed and the perhaps include a more detailed description of the changes after the summary line.

I guess this is the "best practice" reference: (wrong-link-fixed)

https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#commit-message-recommendations

Though I wonder about recommending  `#<issue number>` in the summary because the issue also listed in the body and the commit history does not suggest this is being done consistently.

@HannesWell  Is the best practice description above still the up-to-date recommended best practice?

**Comment by jukzi** [comment]
This repository has a documentation in https://github.com/eclipse-platform/.github/blob/main/CONTRIBUTING.md#commit-message-recommendations

**Comment by DaveCarpeneto** [comment]
Somehow I managed to open a new pull request instead of updating this pull request (apologies - I'm new to GitHub). 

New pull request is https://github.com/eclipse-platform/eclipse.platform.ui/pull/2542 . As such: closing this item

**Pull request closed without merging**